ecopod/sendRawTransactionConditional.md

```diff
@@ -0,0 +1,53 @@
+
+## Verticalize the OP-Stack
+
+Rather than externalize the 4337 mempool, the op-stack could natively offer a UserOp mempool alongside the regular mempool, avoiding conflicts as the sequencer can ensure the bundled UserOps do not conflict with the latest chain state when sealing a new block. However, this adds additional complexity to the stack, where the proposer-builder seperation in 4337 nicely keeps account abstraction out of protocol. A chain operator can still choose to verticalize in proposed solution solution by allowlisting only its own bundler through `eth_sendRawTransactionConditional`, achieving the same outcome.
+
+
+# Proposed Solution
+
+1. Implement `eth_sendRawTransactionConditional` as described in the [spec](https://notes.ethereum.org/@yoav/SkaX2lS9j). A draft implementation for this [exists](https://github.com/ethereum/go-ethereum/compare/master...tynes:go-ethereum:eip4337)today. The conditional is applied on submission and tx inclusion.
+2. Introduce additional validation.
+    * **Exclusive 4337 Entrypoint Support**: We assert `tx.to() == entrypoint_contract_address` to ensure that this endpoint isn't used for other usecases. Making it easy to rollback if deprecated in the future due to native AA or a better solution to this problem.
+    * **Runtime Shutoff**: The endpoint can be shutoff by the sequencer to address issues. As long a single bundler supports a fallback to `eth_sendRawTransaction`, 4337 liveness should remain OK.
+    * **Global Rate Limit**: Rate limit how many conditional txs can be buffered at any given time in the mempool.
+    * **Authentication**: Requests must be authenticated, flashbot-style, with a self-identified keypair. This enables the development of two authorization policy modules.
+        1. _Allowlist_: The sequencer can verticalize their chain with a bundler sidecar or have specific partners allowed to send conditional entrypoint txs.
+        2. _Local Rate Limiting_: The authenticated caller has flashbots-style [reputation](https://docs.flashbots.net/flashbots-auction/advanced/reputation) computed based on the success rate of conditional txs, enabling permisionless bundler participation
+
+The endpoint MUST be authenticated but the authorization policies may not be needed to permissionlessly launch this feature if a global rate limit with a runtime shut off is sufficient. A recommendation is to have the allowlist policy implemented with the keys of known bundlers registered, such that it is ready to enable. As this endpoint is used in production, additional policies like the local rate limits can be iterated on and implemented on a as-needed basis.
```

Comment:
I would recommend on making the initial scope very obvious and then have a section for future iterations. It could be framed like "initial scope, these things can go wrong, if x goes wrong this is solution, if y goes wrong this is the solution..."